files/image_config/procdockerstats/procdockerstats

```diff
@@ -0,0 +1,211 @@
+import re
+import swsssdk 
+from datetime import datetime
+
+VERSION = '1.0'
+
+SYSLOG_IDENTIFIER = "procdockerstats"
+
+REDIS_HOSTIP = "127.0.0.1"
+
+# ========================== Syslog wrappers ==========================
+def log_info(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
+def log_warning(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_WARNING, msg)
+    syslog.closelog()
+
+def log_error(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+# ========================== ProcessDocker class ==========================
+class ProcDockerStats:
+
+    def __init__(self):
+        self.state_db = swsssdk.SonicV2Connector(host=REDIS_HOSTIP)
+        self.state_db.connect("STATE_DB") 
+    
+    def run_command(self, cmd):
+        proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+        (stdout, stderr) = proc.communicate()
+        if proc.returncode != 0:
+        log_error("Error running command '{}'".format(cmd))
+        else:
+            return stdout
+
+    def format_docker_cmd_output(self, cmdout):
+        lines = re.split("\n", cmdout)
+        keys = re.split("   +", lines[0])
+        dict1 = dict()
+        dict_list = []
+        for item in lines[1:]:
+            values1 = re.split("   +", item)
+            dict1 =  dict(zip(keys, values1))
+            dict_list.append(dict1)
+        formatted_dict = self.create_docker_dict(dict_list)
+        return formatted_dict
+
+    def format_process_cmd_output(self, cmdout):
+        lines = re.split("\n", cmdout)
+        keys = re.split(" +", lines[0])
+        keylist = list(filter(None, keys))
+        dict1 = dict()
+        dict_list = []
+        for item in lines[1:]:
+            values1 = re.split(" +", str(item))
+            # To remove extra space before UID
+            val = list(filter(None, values1))
+            # Merging extra columns created due to space in cmd ouput
+            val[8:] = [''.join(val[8:])] 
+            dict1 = dict(zip(keylist, val))
+            dict_list.append(dict1)
+        return dict_list
+
+    def convert_to_bytes(self, value):
+        u = re.search('[a-zA-Z]+', value)
+        v = float(filter(str.isdigit, value))
+        unit = u.group(0)
+        b = 'B'
+        kb = 'KB'
+        mb = 'MB'
+        mib = 'MiB'
+        gib = 'GiB'       
+        if unit.lower() == b.lower():
+            return v
+        elif unit.lower() == kb.lower():
+            v1 = v * 1000
+            return v1
+        elif unit.lower() == mb.lower():
+            v1 = v * 1000000
+            return v1
+        elif unit.lower() == mib.lower():
+            v1 = v * 1048576
+            return v1
+        elif unit.lower() == gib.lower():
+            v1 = v * 1073741824
+            return v1
```

Comment:
Ensure that v1 is a rounded integer value (data can be lost when the value is converted to, say GiB for display then converted back to bytes here). You can simply round downto the nearest integer to play it safe. It will rarely be exact, but it will be close enough.

Reply:
You only added rounding for the return value for the GiB case. Please round all possible return values from this function.

Also, you should probably also convert the return values to integers to ensure all return values are of the same type.